lib/types/request.d.ts

```diff
@@ -296,7 +303,12 @@ export type ReqRef = Partial<Record<keyof ReqRefDefaults, unknown>>;
 /**
  * Utilities for merging request refs and other things
  */
-export type MergeType<T extends object, U extends object> = Omit<T, keyof U> & U;
+export type MergeType<T, U> = {
+    [K in keyof T]: K extends keyof U
+        ? U[K]
+        : T[K];
+};
```

Comment:
`U` would be what you're extending from. Hapi only uses `T`, which is `ReqRefDefaults` in the latter utility type, so any extra properties you might add in the reference type `U` are unused if they're not the same as `T`. This makes intersecting unnecessary.

Does it give you any issues?

Reply:
Nope, but MergeType as a generic type can be abused with a mismatch of keys. If it's not doing an actual merge of keys, maybe it should be named otherwise.

Reply:
Agreed. What name do you suggest? It should only care about objects related to hapi internal types. Anything else would be unused. `ApplyLeft` ?